app/src/processing/app/SketchController.java

```diff
@@ -846,8 +846,8 @@ private static String checkName(String origName) {
 
     if (!newName.equals(origName)) {
       String msg =
-        tr("The sketch name had to be modified. Sketch names can only consist\n" +
-          "of ASCII characters and numbers and be less than 64 characters long.");
+        tr("The sketch name had to be modified. Sketch names can only consist of\n" +
+          "ASCII characters and numbers, no spaces, and be less than 64 characters long.");
```

Comment:
Is this really correct? ASCII characters would include all 128 defined ASCII characters, but the actual limits are smaller: https://github.com/per1234/Arduino/blob/9edfa14531a9d184e36cc5761d4b18984daa3d1b/arduino-core/src/processing/app/BaseNoGui.java#L853-L857

Perhaps this message should just spell out the requirements exactly: A letter or number, followed by letters, numbers, dashes, dots and underscores (underscores are not listed in the whitelist, but anything else is replaced by an underscore, making the underscore implicitly allowed). Maximum length is 63 characters.

Reply:
Great idea. Thanks! I have implemented your suggestion via a force push. I ended up just squashing everything into a single commit since it ends up being a complete rewrite of both messages, rather than changing some specific wording. I like the use of the same wording in both messages, which should make any future changes to them easier. I suppose it would be best practices to just share the same string between the two so they never again get out of sync? Anyway, that's beyond my skills.